### PR TITLE
Conan with Jenkins & docker

### PR DESCRIPTION
**Pull request by danimtb**

### CODE REVIEW

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+package creation. This could be extended to the creation of libraries, bumping of package versions of a of downstream dependencies whenever
+there is a new release upstream, running package tests...
+
+You can also create your own Docker image with the tools you want. Check our
+[conan-docker-tools repository](https://github.com/conan-io/conan-docker-tools) to take some inspiration from the *Dockerfiles*.
+
+We are also aware of some improvements of this kind of set up, for example:
+- The Docker image is pulled only the first time the job runs. It would much more convenient to pull it every time the way
+  [conan-package-tools](https://github.com/conan-io/conan-package-tools#using-docker) does.
+- There are some issues with the Jenkin Artifactory plug-in and ``CONAN_USER_HOME`` not fully working in all cases.
```
Comment (danimtb) [inline _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: Mention here the conan issue(s) related?

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            #ifdef WIRINGPI
+                digitalWrite(0, LOW);
+                delay(500);
+            #endif
+        }
+        return 0;
+    }
+
+As you can see it is a quite simple application with some extra defines to avoid the WirignPi library in case it is not needed. We recently
+included (wiringpi/2.46@conan/stable)[https://bintray.com/conan-community/conan/wiringpi%3Aconan] in conan-center so we will use it as an
```
Comment (danimtb) [inline _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: WiringPi library needs to be included in conan-center before this is pusblished

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown (hunk 1 above)
Comment (lasote) [reply _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: I would remove that paragraph.

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    <img src="{{ site.url }}/assets/post_images/2018-04-23/artifactory.png" align="center"
+    width="600" alt="BlinkApp package for armv7 uploaded to Artifactory"/>
+</p>
+
+The multibranch approach has also other benefits, as you can customize your *Jenkinsfile* in case you have a branch for testing and another
+one stable, as you can change the ``user/channel`` provided in the ``conan create`` step.
+
+Any other approach is also valid, for example, you could trigger the build with a [GitHub hook](https://support.cloudbees.com/hc/en-us/articles/224543927-GitHub-Integration-Webhooks) and have the same behavior as if it was a CI build
+running in Travis CI or AppVeyor.
+
```
Comment (lasote) [inline _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: I would mention what can be done now with the uploaded packages. Not how, but possibilities, deployment, etc

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown (hunk 1 above)
Comment (danimtb) [reply _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: the improvements section you mean, right?

Comment (lasote) [reply _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: yes. from "We are also"

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    width="600" alt="Configuration of the Jenkins Artifactory Plug-in"/>
+  </p>
+
+- **Docker**: Of course, we will be using Docker so we need to have it installed in the CI machine too.
+
+      $ apt-get install docker-ee
+      $ Docker -v
+      > Docker version 1.13.1, build 092cba3
+
+    Check the [isntallation steps](https://docs.docker.com/install/) for other platforms.
```
Comment (memsharded) [inline _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: typo ``isntallation``

File: _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            stage("Upload packages") {
+                String command = "upload BlinkApp* --all -r ${serverName} --confirm"
+                def b = client.run(command: command)
+                b.env.collect()
+                server.publishBuildInfo b
+            }
+        }
+    }
+
+So this is where the magic starts.
```
Comment (memsharded) [inline _posts/2018-04-23-Continuous-integration-for-C-C++embedded-devices-with-jenkins-docker-and-conan.markdown]: No magic

**Pull request merged**